test-framework/test-suites/integration/tests/report/test_report_host_bootfile.py

```diff
@@ -0,0 +1,145 @@
+import pytest
+
+@pytest.mark.usefixtures('revert_database')
+class TestReportHostBootfile:
+	def test_report_host_bootfile_on_fresh_install(self, host):
+		result = host.run('stack report host bootfile')
+		assert result.rc == 0
+
+	def test_report_host_bootfile_for_host_with_full_IP_config(self, host):
+		result = host.run('stack add host backend-1-1')
+		assert result.rc == 0
+		result = host.run('stack add network public address=192.172.0.0 mask=255.255.255.0 pxe=True')
+		assert result.rc == 0
+		result = host.run('stack add host interface backend-1-1 interface=eth1 ip=192.172.0.2 network=public default=True mac=52:54:00:6e:50:91')
+		assert result.rc == 0
+
+		result = host.run('stack report host bootfile')
```

Comment:
I'd like to see a couple short comments in this function, next to each block.

"a host with valid networking info produces a bootfile without error."
"a host with valid networking but without boot action set is skipped without error."